pkg/proxy/ipvs/proxier.go

```diff
@@ -856,13 +856,20 @@ func CleanupLeftovers(ipvs utilipvs.Interface, ipt utiliptables.Interface, ipset
 			encounteredError = true
 		}
 	}
+
 	// Delete dummy interface created by ipvs Proxier.
 	nl := NewNetLinkHandle()
-	err := nl.DeleteDummyDevice(DefaultDummyDevice)
-	if err != nil {
-		glog.Errorf("Error deleting dummy device %s created by IPVS proxier: %v", DefaultDummyDevice, err)
-		encounteredError = true
+	found, err := nl.IsDeviceExist(DefaultDummyDevice)
+	if found {
+		err := nl.DeleteDummyDevice(DefaultDummyDevice)
+		if err != nil {
+			glog.Errorf("Error deleting dummy device %s created by IPVS proxier: %v", DefaultDummyDevice, err)
+			encounteredError = true
+		}
+	} else {
+		glog.V(2).Info("no need to delete dummy device: %v", err)
```

Comment:
the err here is nil?

Reply:
some part of log in my local, i think the log is normal:
```shell
I0126 09:34:51.690608  102055 proxier.go:498] minSyncPeriod: 0s, syncPeriod: 30s, burstSyncs: 2
I0126 09:34:51.690708  102055 server_others.go:173] Tearing down inactive rules.
I0126 09:34:51.701546  102055 proxier.go:870] no need to delete dummy device: Link not found
E0126 09:34:51.712708  102055 proxier.go:838] Failed to execute iptables-restore for nat: exit status 1 (iptables-restore: line 7 failed
)
I0126 09:34:51.719915  102055 server.go:426] Version: v1.10.0-alpha.1.983+68b9d491d78148-dirty
```
WDYT? sorry, may be my misunderstand.

Reply:
I mean why not log the error first. For example,

```
if err != nil {
  ...
} else {
  ...
}
```

Seems more straightforward?